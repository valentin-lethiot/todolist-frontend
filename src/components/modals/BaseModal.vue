<script setup lang="ts">
import Loader from '@/components/loaders/Loader.vue'

type Props = {
  isLoading: boolean
}
defineProps<Props>()
</script>

<template>
  <div id="create-task-modal" class="absolute w-screen h-screen top-0 left-0 bg-gray-900/40 z-50">
    <div class="w-full h-full flex flex-col justify-center items-center p-8">
      <div class="bg-gray-800 rounded-lg p-4 shadow-xl w-full max-h-2/3">
        <div v-if="isLoading" class="flex justify-center items-center h-full w-full">
          <Loader />
        </div>

        <slot v-else />
      </div>
    </div>
  </div>
</template>
