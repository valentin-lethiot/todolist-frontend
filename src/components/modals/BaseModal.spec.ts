import { describe, beforeEach, expect, it } from 'vitest'
import BaseModal from './BaseModal.vue'
import { mount, VueWrapper } from '@vue/test-utils'

describe('BaseModal', () => {
  let wrapper: VueWrapper

  const defaultProps = {
    isLoading: false,
  }

  beforeEach(() => {
    wrapper = mount(BaseModal, {
      props: defaultProps,
      slots: {
        default: '<div>Modal Content</div>',
      },
    })
  })

  describe('render', () => {
    it('should render the modal', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('should render when isLoading is true', async () => {
      await wrapper.setProps({ isLoading: true })
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
