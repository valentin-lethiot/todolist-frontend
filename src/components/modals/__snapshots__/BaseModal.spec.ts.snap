// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseModal > render > should render the modal 1`] = `
"<div id="create-task-modal" class="absolute w-screen h-screen top-0 left-0 bg-gray-900/40 z-50">
  <div class="w-full h-full flex flex-col justify-center items-center p-8">
    <div class="bg-gray-800 rounded-lg p-4 shadow-xl w-full max-h-2/3">
      <div>Modal Content</div>
    </div>
  </div>
</div>"
`;

exports[`BaseModal > render > should render when isLoading is true 1`] = `
"<div id="create-task-modal" class="absolute w-screen h-screen top-0 left-0 bg-gray-900/40 z-50">
  <div class="w-full h-full flex flex-col justify-center items-center p-8">
    <div class="bg-gray-800 rounded-lg p-4 shadow-xl w-full max-h-2/3">
      <div class="flex justify-center items-center h-full w-full">
        <div style="height: 200px; width: 200px;"><canvas style="height: 100%; width: 100%;"></canvas></div>
      </div>
    </div>
  </div>
</div>"
`;
